pages/management/SettingsHandler.php

```diff
@@ -53,6 +55,27 @@ public function __construct()
     public function workflow($args, $request)
     {
         parent::workflow($args, $request);
+
+        $templateMgr = TemplateManager::getManager($request);
+
+        $context = $request->getContext();
+        $dispatcher = $request->getDispatcher();
+
+        $contextApiUrl = $dispatcher->url($request, PKPApplication::ROUTE_API, $context->getPath(), 'contexts/' . $context->getId());
+
+        $locales = $context->getSupportedFormLocaleNames();
+        $locales = array_map(fn (string $locale, string $name) => ['key' => $locale, 'label' => $name], array_keys($locales), $locales);
```

Comment:
I'm happy with this as-is. A suggestion would be to move this code into a helper method in `ManagementHandler` in pkp-lib, and use that, just to minimize code duplication. So in `ManagementHandler` it would be:

```php
protected function getContextApiUrl(): string
{
    //
}
protected function getFormLocales(): array
{
    //
}
```

Then you can use those methods in the parent and child classes without duplicating the code. Making the methods `protected` means that if/when we refactor the code, we can be confident when the helper methods are no longer used. The code editor will even tell us when the methods are not used.

But that's just a suggestion. If you want to go ahead with it as-is, no problem.